crates/wast/src/ast/types.rs

```diff
@@ -246,22 +252,102 @@ impl<'a> Parse<'a> for FunctionType<'a> {
     }
 }
 
+/// A struct type with fields.
+#[derive(Clone, Debug)]
+pub struct StructType<'a> {
+    /// The fields of the struct
+    pub fields: Vec<StructField<'a>>,
+}
+
+impl<'a> StructType<'a> {
+    fn finish_parse(&mut self, parser: Parser<'a>) -> Result<()> {
+        while parser.peek2::<kw::field>() {
```

Comment:
Should this be `while !parser.is_empty()`? Or are there other things parsed elsewhere?

Reply:
No there's not, remnants of a copy-paste probably.